### PR TITLE
Upgrade to Ubuntu 22.04 and Python 3.10 (PP-506)

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Update the base image that we use to Ubuntu 22.04. This will let us deploy using Python 3.10. Opening the way for us to drop support for Python 3.8 and 3.9.

## Motivation and Context

Right now the CM runs on Python 3.8. We should upgrade to a newer Python version, so that we can eventually drop Python 3.8. Python 3.8 will be unsupported by 14 Oct 2024, but I’d like to move off it it long before then.

## How Has This Been Tested?

- Tests in CI
- Built image and ran it locally, everything seemed okay

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by tdilauro — APPROVED** [review]
Looks good!

File: docker/ci/test_webapp.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ check_service_status "$container" /etc/service/nginx
 check_service_status "$container" /etc/service/uwsgi
 
 # Wait for UWSGI to be ready to accept connections.
-timeout 120s grep -q 'WSGI app .* ready in [0-9]* seconds' <(docker logs "$container" -f 2>&1)
+timeout 240s grep -q 'WSGI app .* ready in [0-9]* seconds' <(docker logs "$container" -f 2>&1)
```
Comment (tdilauro) [inline docker/ci/test_webapp.sh:20]: Also, wow! Is it really taking four minutes to spin up?

Comment (jonathangreen) [reply docker/ci/test_webapp.sh:20]: No, looking at the logs where it failed, it seems like it was taking about 130s, just beyond the previous timeout. But I figured I'd give it a bit of a safety factor and just double what we had before.

**Comment by jonathangreen** [comment]
I'm holding off on merging this until I cut the new staging release, so we have a bit of time to test this out on minotaur before it hits staging.

**Pull request merged**